entrypoint.sh

```diff
@@ -41,43 +57,55 @@ chown nginx:nginx /var/cache/nginx
 mkdir -p /var/tmp/nginx
 chown nginx:nginx /var/tmp/nginx
 
+#create vhost directory
+mkdir -p /etc/nginx/vhosts/
+
+# Copy nginx main conf file.
+cp templates/nginx.conf /etc/nginx/nginx.conf
```

Comment:
Yes, I've been wondering about that. Though you can still customize the way nginx globally behave by editing the nginx.conf file, you can't tune the behaviour on a vhost basis, since I chose to use the same template.

I think that we could look if there is a `templates/${DOMAIN}.conf` file that exist and use it if so, and if not use the default file. 
That would allow doing some tuning on a vhost/domain basis, the drawback being that you loose some automation (but if you need specific options on a specific vhost you won't have automation anyway, right?)

A solution to preserve compatibility with the previous `nginx.conf` files could be to check if there is a `server{}` in the file, if so, we would ignore the template file and do the `sed` job on that file.